tasks/facts_diff.rb

```diff
@@ -0,0 +1,37 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+# frozen_string_literal: true
+
+require 'json'
+require 'puppet'
+
+params = JSON.parse(STDIN.read)
+require_relative File.join(params['_installdir'], 'puppet_agent', 'files', 'rb_task_helper.rb')
+
+module PuppetAgent
+  class FactsDiff
+    include PuppetAgent::RbTaskHelper
+
+    def run
+      unless puppet_bin_present?
```

Comment:
This also seems like it should verify the agent version is `6.x` so that the command behaves as expected instead of returning facts for the target (which is what happens with 5.x versions). It's actually also unclear to me if this subcommand is supported in 7.x - it's only documented in [6.x documentation](https://puppet.com/docs/puppet/7.4/man/facts.html), but seems to work in 7.x?

Reply:
The `puppet facts diff` command is only available in puppet 6(>= 6.20.0). Added a check for that